frontend/src/pages/ArtifactDetails.tsx

```diff
@@ -95,6 +95,26 @@ export default class ArtifactDetails extends Page<{}, ArtifactDetailsState> {
     return this.load();
   }
 
+  public componentDidUpdate(
+    prevProps: Readonly<{} & PageProps>,
+    prevState: Readonly<ArtifactDetailsState>,
+    snapshot?: any,
+  ): void {
+    if (
+      this.props.match.params[RouteParams.ID] === prevProps.match.params[RouteParams.ID] &&
+      prevState.selectedTab !== ArtifactDetailsTab.LINEAGE_EXPLORER
```

Comment:
so, any component update triggered when this is in lineage_explorer will cause a reload. I think that's a dangerous behavior.

e.g.
1. go to lineage tab
2. go to overview tab

In this PR, the page will reload, https://screencast.googleplex.com/cast/NDk3ODg1NzI2NDI4MzY0OHxkZmQ2YTU2Mi1hOA.
Before this PR, the tab will switch smoothly, and data is already present.

If there are any top tree updates that accidentally comes in, this page will reload.